libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -449,6 +449,28 @@ AP_GPS::update(void)
 }
 
 /*
+  GPS time of week in days, hours, minutes, seconds, milliseconds
+*/
+void AP_GPS::time_week_utc(uint8_t instance, uint16_t &day, uint16_t &hour, uint16_t &min, uint16_t &sec, uint16_t &ms) const
```

Comment:
Please see my above note as to why this should all move out of the GPS library all together. On first fix we move the system time to be based on GPS time, but that time will keep counting forwards even if we lose the GPS for some reason.